### PR TITLE
handle mesos tasks having ids that are substrings of each other

### PR DESCRIPTION
**Pull request by tamarrow**
Instead of crashing, if looking for a task by a fltr, if a task is an
exact match, return that task.

### CODE REVIEW

File: dcos/mesos.py [hunk 1]
```diff
@@ -346,10 +347,15 @@ def slave(self, fltr):
             raise DCOSException('No slave found with ID "{}".'.format(fltr))
 
         elif len(slaves) > 1:
-            matches = ['\t{0}'.format(slave['id']) for slave in slaves]
+
+            exact_matches = [s for s in slaves if s['id'] == fltr]
+            if len(exact_matches) == 1:
+                return exact_matches[0]
+
```
Comment (jsancio) [inline dcos/mesos.py]: Should we add an `else` to make it clear?

**Comment by jsancio** [comment]
LGTM. One minor comment. Can we write a test for this?

**Comment by tamarrow** [comment]
thanks! updated with a unit test.

**Comment by jsancio** [comment]
:shipit:

**Comment by jsancio** [comment]
Fixes #660

**Pull request merged**